src/fsverity/mod.rs

```diff
@@ -19,3 +25,57 @@ impl FsVerityHashValue for Sha512HashValue {
     const ALGORITHM: u8 = 2;
     const EMPTY: Self = [0; 64];
 }
+
+pub fn sha256_from_str(s: &str) -> Result<Sha256HashValue> {
+    let mut r: Sha256HashValue = Default::default();
+    hex::decode_to_slice(s, r.as_mut_slice())?;
+    Ok(r)
+}
+
+/// A verity comparison failed.
+#[derive(Error, Debug)]
+pub enum CompareVerityError {
+    #[error("failed to read verity")]
+    Measure(#[from] MeasureVerityError),
+    #[error("fsverity is not enabled on target file")]
+    VerityMissing,
+    #[error("Expected digest {expected} but found {found}")]
+    DigestMismatch { expected: String, found: String },
+}
+
+/// Compare the fsverity digest of the file versus the expected digest.
+pub fn compare_verity<F: AsFd, H: FsVerityHashValue>(
+    fd: F,
+    expected: &H,
+) -> Result<(), CompareVerityError> {
```

Comment:
```suggestion
pub fn compare_verity(fd: impl AsFd, expected: impl FsVerityHashValue) -> Result<(), CompareVerityError> {
```

Reply:
I briefly tried that but I ran into type inference errors on the invocation of `fs_ioc_measure_verity`